### ISSUE TITLE
Variable expressions coerce `NaN` to `undefined`

### ISSUE THREAD

**Issue by chrisdickinson** [thlorenz/scriptie-talkie#1]
[an example](http://thlorenz.github.io/scriptie-talkie/?code=var%20a%20%3D%201%20%2B%20void%20%2B%201%3B%0Atypeof%20a%20%3D%3D%3D%20'undefined'%0A%0Avar%20b%20%3D%201%20%2B%20undefined%20%2B%201%3B%0Atypeof%20b%20%3D%3D%3D%20'undefined'%0A%0A%2F%2FNaN%0A%0Ax%20%3D%20NaN%20%2B%201%0Atypeof%20x%0A%0Atypeof%20NaN%0A%0Atypeof%20(NaN%20%2B%20%22%22\)%0A%0Atypeof%20(1%20%2B%20void%20%2B%201\)%0A%0A). 

Basically, `typeof(1 + undefined + 1)` will output `"number"` (as expected -- the value is `NaN`), but:

``` javascript
x = 1 + undefined + 1
typeof(x)
```

will output `"undefined"` unexpectedly.

**Comment by thlorenz** [comment]
As I suspected, this is only broken in the browser: https://github.com/thlorenz/scriptie-talkie/commit/a490ce91a81d991385c660f102a0bd77ec4508da

It's a bit more hacky there, will try to figure out how to fix it everywhere but the most important part is that it works server side since it will be part of replpad and the browser thing is merely a neat demo.

**Comment by thlorenz** [comment]
This issue is fixed now, just click the link again.
However all kind of `~NaN`s appear although nothing changed. This is due to a bug in deep-equal `NaN !== NaN` ;) which I have a [PR](https://github.com/substack/node-deep-equal/pull/5) for. Just waiting for @substack to pull it in and all things should be dandy.

**Comment by thlorenz** [comment]
Printing multiple `NaN`s issue is now also resolved.
Closing.
